provider/alibabacloud/alibaba_cloud.go

```diff
@@ -373,24 +374,22 @@ func (p *AlibabaCloudProvider) groupRecords(records []alidns.Record) (endpointMa
 func (p *AlibabaCloudProvider) records() ([]alidns.Record, error) {
 	log.Infof("Retrieving Alibaba Cloud DNS Domain Records")
 	var results []alidns.Record
-
-	if len(p.domainFilter.Filters) == 0 {
-		domainNames, tmpErr := p.getDomainList()
-		if tmpErr != nil {
-			log.Errorf("AlibabaCloudProvider getDomainList error %v", tmpErr)
-			return results, tmpErr
-		}
-		for _, tmpDomainName := range domainNames {
-			tmpResults, err := p.getDomainRecords(tmpDomainName)
+	hostedZoneDomains, err := p.getDomainList()
+	if err != nil {
+		return results, fmt.Errorf("getting domain list: %w", err)
+	}
+	if !p.domainFilter.IsConfigured() {
+		for _, zoneDomain := range hostedZoneDomains {
+			domainRecords, err := p.getDomainRecords(zoneDomain)
 			if err != nil {
-				log.Errorf("AlibabaCloudProvider getDomainRecords %s error %v", tmpDomainName, err)
-				continue
+				log.Errorf("AlibabaCloudProvider getDomainRecords %q error %v", zoneDomain, err)
+				return nil, err
```

Comment:
Don't both log and return an error.

```suggestion
				return nil, fmt.Errorf("getDomainRecords %q: %w", zoneDomain, err)
```